### PR TITLE
Route refresh: added `incidents`

### PR DESCRIPTION
**Pull request by RingerJK**
~depends on https://github.com/mapbox/mapbox-java/pull/1421~

require NN API: update incidents (⚠️ landed in 99.0.0)

### CODE REVIEW

**Comment by VysotskiVadim** [comment]
Please invalidate incidents in the same way as annotations if https://github.com/mapbox/mapbox-navigation-android/pull/5767 lands first.

File: build.gradle [hunk 1]
```diff
@@ -58,7 +58,7 @@ allprojects {
     // with Navigation Native forced to be some snapshot version
     // if you need to use snapshots while development, just set `addSnapshotsRepo` to true manually
     def addSnapshotsRepo = project.hasProperty('ALLOW_SNAPSHOT_REPOSITORY') ? project.property('ALLOW_SNAPSHOT_REPOSITORY') : (System.getenv("ALLOW_SNAPSHOT_REPOSITORY")?.toBoolean() ?: false)
-    if (addSnapshotsRepo) {
+    if (true) {
```
Comment (RingerJK) [inline build.gradle:61]: uses to allow snapshots only, will be reverted when the next mapbox-java will be released

Comment (Zayankovsky) [reply build.gradle:61]: v6.5.0-beta.5 should be available already.

File: libnavigation-base/src/test/java/com/mapbox/navigation/base/internal/route/NavigationRouteExTest.kt [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+import com.mapbox.navigator.RouterOrigin
+import io.mockk.every
+import io.mockk.mockk
+import org.junit.Assert.assertEquals
+import org.junit.Test
+
+class NavigationRouteExTest {
+
+    @Test
+    fun `extension NavigationRoute refreshRoute`() {
```
Comment (VysotskiVadim) [inline libnavigation-base/src/test/java/com/mapbox/navigation/base/internal/route/NavigationRouteExTest.kt:18]: No actions required, just a feedback from me. I find this "parametrised" tests hard to read. I spent 20 minutes and didn't really get all the scenarios. Usually if you write a test per scenario, tests are getting smaller and easier to understand

**Review by LukasPaczos — COMMENTED** [review]
@RingerJK we also need to extract the road objects from `RouteInterface#routeInfo#alerts` after calling `Navigator#refreshRoute` and deliver these refreshed alerts in the `RoadObjectsOnRouteObserver`.

File: libnavigation-base/src/test/java/com/mapbox/navigation/base/internal/route/NavigationRouteExTest.kt (hunk 2 above)
Comment (RingerJK) [reply libnavigation-base/src/test/java/com/mapbox/navigation/base/internal/route/NavigationRouteExTest.kt:18]: thanks for the feedback, parametrized tests are helpful for different cases in the same scenario (like here), not setting up an environment for each test. I'd add more docs for current tests, to make it readable

File: CHANGELOG.md [hunk 3]
```diff
@@ -8,6 +8,7 @@ Mapbox welcomes participation and contributions from everyone.
 - Added `TollCollection#name` field which contains a name of the toll collection point, when available. [#5784](https://github.com/mapbox/mapbox-navigation-android/pull/5784)
 - Added `ReplayRouteOptions#frequency` to allow adjusting the simulated location frequency. [#5724](https://github.com/mapbox/mapbox-navigation-android/pull/5724)
 - Added `MapboxReplayer#eventRealtimeOffset` to allow adjusting simulated locations timestamps for more accurate replays. [#5748](https://github.com/mapbox/mapbox-navigation-android/pull/5748)
+- Added update of `incidents` for route refresh. For now `incidents` are refreshed along a route. [#5749](https://github.com/mapbox/mapbox-navigation-android/pull/5749)
```
[the remark this answers is not in the document]
Comment (LukasPaczos) [reply CHANGELOG.md:11]: Please also add an instrumentation test for this in `com.mapbox.navigation.instrumentation_tests.core.RouteRefreshTest`.

[the remark this answers is not in the document]
Comment (RingerJK) [reply CHANGELOG.md:11]: We usually start changelog lines from Added/Updated/Removed. I'd change to align 
```suggestion
- Updated feature `route refresh`: now also supports refreshing `incidents` along the route found in `RouteLeg#incidents` and `RoadObjectsOnRouteObserver`. [#5749](https://github.com/mapbox/mapbox-navigation-android/pull/5749)
```

**Comment by RingerJK** [comment]
- added refresh route tests to `com.mapbox.navigation.instrumentation_tests.core.RouteRefreshTest` (in the same test added check `RoadObjectsOnRouteObserver` data);
- removed refresh route tests from `com.mapbox.navigation.navigator.internal.NavigatorTest` as it duplicates with the above point;
- added description to parametrized test wrappers of `com.mapbox.navigation.base.internal.route.NavigationRouteExTest`;
- fixed changelog.

**Pull request merged**